resources/type-system/strict-inference.md

```diff
@@ -171,7 +171,11 @@ type cannot be inferred from downwards inference (in the case of a function
 literal) is considered an inference failure. A function literal's parameter
 types are commonly inferred when assigning the literal to a variable typed with
 a typedef, or when passing the literal as an argument whose corresponding
-parameter is  function-typed.
+parameter is function-typed.
+
+Parameters named `_`, `__`, etc. do not represent an inference failure. Such
+names traditionally represent variables which are ignored, and as such, their
+type, `dynamic` or not, is not important.
```

Comment:
Do we want to specify that they are only allowed (not an inference failure) if they are, in fact, not referenced within the function? Not doing so leaves a hole, but it's probably uncommon enough that it might not be worth specifying. On the other hand, the analysis isn't that hard, so it might be worth it.

Reply:
Goooood question. I'd say that case of using a variable named `_` is quite uncommon (in fact, we could add a lint for it 😄). I'm ok not specifying anything about the use of `_`.

Reply:
How does a variable "represent" an inference failure? The word "represent" is not used before in this document.

Maybe:

> Parameters named \`_\`, \`__\`, etc., are not considered inference failures if they have no type, 
> as long as they are not referenced inside the body of the function.

Or mix it into the previous paragraph:

> A function parameter declared ....
> ... is considered an inference failure, except if the variable is named \`\_\`, \`\_\_`, etc., 
> and the variable is not referenced from inside the function body.

(Or just ignore it for any variable which is not used).

Reply:
I like your last suggestion. I've written text to follow this idea.